analyzer_test.go

```diff
@@ -118,532 +120,55 @@ func testAnalyzerBuilder(platformAPI string) func(t *testing.T, when spec.G, it
-						h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "metadata.buildpack", "launch-cache-not-in-app.sha"))
-					})
-
-					it("does not restore cache=false layer metadata", func() {
-						h.SkipIf(t, api.MustParse(platformAPI).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not restore layer metadata")
-
-						_, err := analyzer.Analyze()
-						h.AssertNil(t, err)
-
-						h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "metadata.buildpack", "cache-false.toml"))
-						h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "metadata.buildpack", "cache-false.sha"))
-					})
-
-					it("restores escaped buildpack layer metadata", func() {
-						h.SkipIf(t, api.MustParse(platformAPI).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not restore layer metadata")
-
-						_, err := analyzer.Analyze()
-						h.AssertNil(t, err)
-
-						path := filepath.Join(layersDir, "escaped_buildpack_id", "escaped-bp-layer.toml")
-						got := h.MustReadFile(t, path)
-						want := "[metadata]\n  escaped-bp-layer-key = \"escaped-bp-layer-value\""
-						unsetFlags := "[types]"
-
-						h.AssertStringContains(t, string(got), want)
-						h.AssertStringDoesNotContain(t, string(got), unsetFlags) // The [types] table shouldn't exist. The build, cache and launch flags are set to false.
-					})
-
-					when("subset of buildpacks are detected", func() {
-						it.Before(func() {
-							analyzer.Buildpacks = []buildpack.GroupBuildpack{{ID: "no.group.buildpack", API: api.Buildpack.Latest().String()}}
-						})
-
-						it("restores layers for detected buildpacks", func() {
-							h.SkipIf(t, api.MustParse(platformAPI).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not restore layer metadata")
-
-							_, err := analyzer.Analyze()
-							h.AssertNil(t, err)
-
-							path := filepath.Join(layersDir, "no.group.buildpack", "some-layer.toml")
-							got := h.MustReadFile(t, path)
-							want := "[metadata]\n  some-layer-key = \"some-layer-value\""
-
-							h.AssertStringContains(t, string(got), want)
-						})
-
-						it("does not restore layers for undetected buildpacks", func() {
-							h.SkipIf(t, api.MustParse(platformAPI).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not restore layer metadata")
-
-							_, err := analyzer.Analyze()
-							h.AssertNil(t, err)
-
-							h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "metadata.buildpack"))
-							h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "escaped_buildpack_id"))
-						})
-					})
-
-					when("buildpack API < 0.6", func() {
-						it.Before(func() {
-							analyzer.Buildpacks = []buildpack.GroupBuildpack{
-								{ID: "metadata.buildpack", API: "0.5"},
-								{ID: "no.cache.buildpack", API: "0.5"},
-							}
-						})
-
-						it("restores app and cache layer metadata and preserves the values of the launch, build and cache flags", func() {
```

Comment:
Do we continue testing it?
I couldn't find this test anywhere.

Reply:
https://github.com/buildpacks/lifecycle/pull/605/files#diff-6834a3c40b637f9d72a33e93feaa8db461b3551f15e6a1fffa1a5c6dd6cd4ccdR144

Reply:
You're right!
I search for `launch=true` instead of `launch = true`.
Thanks!